File: rustfst/src/drawing_config.rs
/// Struct to configure how the FST should be drawn.
#[derive(Debug, Clone, PartialEq)]
pub struct DrawingConfig {
    /// Draw bottom-to-top instead of left-to-right.
    pub vertical: bool,
    /// Set pair (width, height)
    pub size: Option<(f32, f32)>,
    /// Set figure title.
    pub title: String,
    /// Portrait mode (def: landscape).
    pub portrait: bool,
    /// Set minimum separation between ranks (see dot documentation).
    pub ranksep: Option<f32>,
    /// Set minimum separation between nodes (see dot documentation).
    pub nodesep: Option<f32>,
    /// Set fontsize.
    pub fontsize: u32,
    /// Input in acceptor format.
    pub acceptor: bool,
    /// Print/draw transition weights and final weights equal to Weight::ONE.
    pub show_weight_one: bool,
    /// Print/draw transition weights and final weights.
    pub print_weight: bool,
}

impl Default for DrawingConfig {
    fn default() -> Self {
        Self {
            vertical: false,
            size: None,
            title: "".to_string(),
            portrait: false,
            ranksep: None,
            nodesep: None,
            fontsize: 14,
            acceptor: false,
            show_weight_one: true,
            print_weight: true,
        }
    }
}

File: rustfst-python/rustfst/drawing_config.py
from typing import Optional


class DrawingConfig:
    def __init__(
        self,
        acceptor: bool = False,
        title: str = "",
        width: Optional[float] = None,
        height: Optional[float] = None,
        portrait: bool = True,
        vertical: bool = False,
        ranksep: Optional[float] = None,
        nodesep: Optional[float] = None,
        fontsize: int = 14,
        show_weight_one: bool = True,
        print_weight: bool = True,
    ):
        """
        Args:
            acceptor: Should the figure be rendered in acceptor format if possible?
            title: An optional string indicating the figure title.
            width: The figure width, in inches.
            height: The figure height, in inches.
            portrait: Should the figure be rendered in portrait rather than
                landscape?
            vertical: Should the figure be rendered bottom-to-top rather than
                left-to-right?
            ranksep: The minimum separation separation between ranks, in inches.
            nodesep: The minimum separation between nodes, in inches.
            fontsize: Font size, in points.
            show_weight_one: Should weights equivalent to semiring One be printed?
            print_weight: Should weights be printed
        """
        self.acceptor = acceptor
        self.title = title
        self.width = width
        self.height = height
        self.portrait = portrait
        self.vertical = vertical
        self.ranksep = ranksep
        self.nodesep = nodesep
        self.fontsize = fontsize
        self.show_weight_one = show_weight_one
        self.print_weight = print_weight
